### PR TITLE
Update pr checklist to remind updating the changelog

### PR DESCRIPTION
**Pull request by maurolepore**
The changelog is becoming more important as our packages gradually gain more visibility -- to other developers as we increasingly collaborate, and to Deloitte team as we report our monthly progress during the dev-cycle meetings. 

I edit the changelog with [fledge](https://fledge.cynkra.com/) but it's also easy to do manually:

- Edit NEWS.md
- Increment the version number in DESCRIPTION
- Tag the commit with the new version with something like:

```bash
git tag -a <version> -m "Some message"
git push origin <version>
```

----

TODO

- [ ] [Link related issue/PR]([url](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)). 
- [x] Describe the goal of the PR. Avoid details that are clear in the diff.
- [x] Mark the PR as draft.
- [ ] [Include a unit test](https://code-review.tidyverse.org/reviewer/aspects.html#sec-tests).
- [x] Review your own PR in "Files changed".
- [x] Ensure the PR branch is updated.
- [x] Ensure the checks pass.
- [x] Change the status from draft to ready.
- [x] Polish the PR title and description.
- [x] Assign a reviewer.

EXCEPTIONS

- [ ] Slide here any item that you intentionally choose to not do.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -10,9 +12,7 @@ TODO
 - [ ] Ensure the PR branch is updated.
 - [ ] Ensure the checks pass.
 - [ ] Change the status from draft to ready.
-- [ ] Polish the PR title and description.
+- [ ] Polish the PR description to reflect what it ended up doing.
```
Comment (maurolepore) [inline .github/pull_request_template.md:15]: This helps forensic exploration of what happened in each PR.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -10,9 +12,7 @@ TODO
...
+- [ ] Polish the PR title as you'd like others to read it from the git log.
```
Comment (maurolepore) [inline .github/pull_request_template.md:16]: This leaves a readable record in the commit history.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -10,9 +12,7 @@ TODO
...
 - [ ] Assign a reviewer.
-
-EXCEPTIONS
-
-- [ ] Slide here any item that you intentionally choose to not do.
+- [ ] Document user-facing changes in the changelog.
```
Comment (maurolepore) [inline .github/pull_request_template.md:18]: This forces us to think which changes affect users.

**Review by maurolepore — COMMENTED** [review]
Some comments about why I did this changes.

**Review by lindadelacombaz — APPROVED** [review]

**Pull request merged**